public/app/core/components/PasswordField/PasswordField.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { FC, useState } from 'react';
+import { selectors } from '@grafana/e2e-selectors';
+
+import { Input, IconButton } from '@grafana/ui';
+
+export interface Props {
+  autoFocus?: boolean;
+  autoComplete?: string;
+  id?: string;
+  passwordHint?: string;
+}
+
+export const PasswordField: FC<Props> = ({ autoComplete, autoFocus, id, passwordHint, ...props }) => {
+  const [showPassword, setShowPassword] = useState(false);
+
+  return (
+    <Input
+      id={id}
+      autoFocus={autoFocus}
+      autoComplete={autoComplete}
+      {...props}
+      type={showPassword ? 'text' : 'password'}
+      placeholder={passwordHint}
+      aria-label={selectors.pages.Login.password}
+      suffix={
+        <IconButton
+          name={showPassword ? 'eye-slash' : 'eye'}
+          surface="header"
+          onClick={(e) => {
+            e.preventDefault();
+            setShowPassword(!showPassword);
+          }}
+        />
+      }
+    />
+  );
+};
```

Comment:
```suggestion
export const PasswordField: FC<Props> = React.forwardRef<HTMLInputElement, Props>(
  ({ autoComplete, autoFocus, id, passwordHint, ...props }, ref) => {
    const [showPassword, setShowPassword] = useState(false);
    console.log('props', props);
    return (
      <Input
        id={id}
        autoFocus={autoFocus}
        autoComplete={autoComplete}
        {...props}
        type={showPassword ? 'text' : 'password'}
        placeholder={passwordHint}
        aria-label={selectors.pages.Login.password}
        ref={ref}
        suffix={
          <IconButton
            name={showPassword ? 'eye-slash' : 'eye'}
            surface="header"
            onClick={(e) => {
              e.preventDefault();
              setShowPassword(!showPassword);
            }}
          />
        }
      />
    );
  }
);

PasswordField.displayName = 'PasswordField';
```

The `PasswordField` needs to be wrapped in `forwardRef` to be able to accept `ref` from `register`.

Reply:
@Clarity-89, the `Input` component is already wrapped in `forwardRef`, do still need to add one more wrapper here?.
IMO, it is not needed and tested all scenarios and the `register` is working fine

Reply:
If you check the browser console on the login page you'll see the ref warning, adding forwardRef fixes that.

Reply:
thanks, addressed